src/sinks/util/service.rs

```diff
@@ -174,39 +71,72 @@ pub struct TowerRequestConfig<T: ConcurrencyOption = Concurrency> {
     #[serde(skip_serializing_if = "ConcurrencyOption::is_none")]
     pub in_flight_limit: T, // 5
     pub timeout_secs: Option<u64>,             // 60
-    pub rate_limit_duration_secs: Option<u64>, // 1
-    pub rate_limit_num: Option<u64>,           // 5
-    pub retry_attempts: Option<usize>,         // max_value()
+    pub rate_limit_duration_secs: Option<u64>, // 1 day
```

Comment:
I think this could be a bit confusing for users as they'd commonly need to adjust this. Maybe we could just make this `1` by default? With `rate_limit_num` maxed out below this should effectively be unlimited even if this is left at `1`.

Reply:
If possible I'd like to avoid the sleep churn that a low duration here implies. The rate-limit implementation builds sleeps based on the set `Rate`, so the lower the more often futures get woken up. We spend a non-trivial time in vector today servicing sleeps, is my motivation here.

Reply:
Aha, yes, that makes sense as a goal. My concern is just that users will set `rate_limit_num` to, say 5, and be surprised that they are only getting 5 requests per day.

I'm wondering if there is another way we could model this to avoid even introducing the rate limiting layer when no rate limit is being applied. Say, `rate_limit_num` is set to `None`, could we just not apply a rate limit at all rather than defaulting to a very high number to effectively mimic that? Then we could default `rate_limit_duration_secs` to `1` still so, that when they do apply a limit, it is per second, which I think is what users will expect.

Reply:
Right now that we have a rate limit is encoded in the types, so flagging it off unless explicitly enabled is a bigger change than it might seem. I do agree, however, that setting `rate_limit_num` and getting a per-day result is not super desirable. While we could also call this out clearly in the documentation it's still an awfully sharp edge. Seems to me our whole stack here could use some TLC -- why, for instance, do we not have a compression layer installed and rely on sinks to implement it themselves? -- and getting rate limits to an optional thing might be a goal there. 

How about we split the difference some, consider how we're wedge in here? Maybe we make the default duration 1 minute? It's still fairly wide but not absolutely shocking.

Reply:
Said change introduced in 86024a0e75b4f8b83928c71b70497bf2bf820923

Reply:
Would it make a sense to normalize this value, so that if the user specifies a value under, say, `60`, it sets it to `60` and scales up the `rate_limit_num` to compensate?

Reply:
I like the idea. Effectively we always normalize to a per-second value -- or per whatever -- and adjust internally to hit that target? This gets tricky around burst behavior though. Imagine if the user rigs for 10k requests per second, we adjust that to be 60k requests per minute and then the whole quota is consumed in the first second. That would surprise me, anyhow, and I'm not sure how to avoid that situation without introducing configuration to control burst sizes.

Reply:
I know it's gross, but could we mimic what we do for the tracing layers to optionally include this layer?

https://github.com/timberio/vector/blob/cee390c1282f134da4f9bbee396b7eef89b45138/src/trace.rs#L53-L125

This does seem like a short-coming of these builder patterns that return generic types in that it is difficult to conditionally enable certain features.

Reply:
I'd rather it be slightly gross internally, but nice for users than vice-versa.

Reply:
It's possible? Our existing stack is complicated enough that I argue it needs a refresh anyway and we probably need to directly expose how the layer stack is constructed to each sink to allow tailoring. There's other layers we'll want to optionally enable, like compression. We'll be forced to reconsider all this anyway once we jump our throughput goals past 10Mb/s per core. 

My appetite in this PR is to just revert back to a default duration of 1 second, not add more complication to the implementation and note this as a sharp corner we want to address in the future.

Reply:
Revision to 1 second duration done in 8df6bb8a8d2b94e2a14c36b6f8081988957736b8

Reply:
👍 yeah, I am unsure if it is possible to untangle easily. Thanks for bumping back to `1` for now.